File: Basic/9.Write a program to find the factorial of a number/Solution.py
def factorial(n):
    fact = 1
    for i in range(1, n + 1):
        fact *= i
    return fact

if __name__ == "__main__":
    n = int(input("Enter a number: "))

    if n < 0:
        print("Factorial is not defined for negative numbers.")
    else:
        print(f"Factorial of {n} is {factorial(n)}")

File: Basic/9.Write a program to find the factorial of a number/Solution.c
#include <stdio.h>

unsigned long long factorial(int n) {
    unsigned long long fact = 1;
    for (int i = 1; i <= n; i++) {
        fact *= i;
    }
    return fact;
}

int main() {
    int n;

    printf("Enter a number: ");
    scanf("%d", &n);

    if (n < 0) {
        printf("Factorial is not defined for negative numbers.\n");
    } else {
        printf("Factorial of %d is %llu\n", n, factorial(n));
    }

    return 0;
}
